src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/InstanceDiscovery.cs

```diff
@@ -187,11 +187,15 @@ private async Task DiscoverAsync(Uri authority, bool validateAuthority, RequestC
                 // All other unexpected exceptions will still bubble up, as always.
                 if (validateAuthority)
                 {
-                    // hard stop here
-                    throw new AdalException(
-                        (ex.ErrorCode == "invalid_instance")
-                            ? AdalError.AuthorityNotInValidList
-                            : AdalError.AuthorityValidationFailed, ex);
+                    if (Equals(ex.ErrorCode == ErrorCodes.InvalidInstance, StringComparison.OrdinalIgnoreCase))
+                    {
+                        throw new AdalException(
+                            AdalError.AuthorityNotInValidList, 
+                            AdalError.AuthorityValidationFailed, 
```

Comment:
Is this (AuthorityValidationFailed) an actual error message?

Reply:
we turn the invalid_instance into authority validation failed...is that what you mean?

---
In reply to: [356264823](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1682#discussion_r356264823) [](ancestors = 356264823)